Resources/Computing/ARCComputingElement.py

```diff
@@ -291,19 +294,49 @@ def getCEStatus( self ):
     if not result['OK']:
       gLogger.error( 'ARCComputingElement: failed to set up proxy', result['Message'] )
       return result
-    endpoints = [arc.Endpoint( "ldap://" + self.ceHost + "/MDS-Vo-name=local,o=grid",
-                               arc.Endpoint.COMPUTINGINFO, 'org.nordugrid.ldapng')]
-    retriever = arc.ComputingServiceRetriever(self.usercfg, endpoints)
-    retriever.wait() # Takes a bit of time to get and parse the ldap information
-    targets = retriever.GetExecutionTargets()
-    ceStats = targets[0].ComputingShare
-    gLogger.debug("Running jobs for CE %s : %s" % (self.ceHost, ceStats.RunningJobs))
-    gLogger.debug("Waiting jobs for CE %s : %s" % (self.ceHost, ceStats.WaitingJobs))
+
+    # Try to find out which VO we are running for.
+    vo = ''
+    res = getVOfromProxyGroup()
+    if res['OK']:
+      vo = res['Value']
 
     result = S_OK()
-    result['RunningJobs'] = ceStats.RunningJobs
-    result['WaitingJobs'] = ceStats.WaitingJobs
     result['SubmittedJobs'] = 0
+    if (vo == '') :
+      # Presumably the really proper way forward once the infosys-discuss WG comes up with a solution
+      # and it is implemented. Needed for DIRAC instances which use robot certificates for pilots.
+      endpoints = [arc.Endpoint( "ldap://" + self.ceHost + "/MDS-Vo-name=local,o=grid",
+                               arc.Endpoint.COMPUTINGINFO, 'org.nordugrid.ldapng')]
+      retriever = arc.ComputingServiceRetriever(self.usercfg, endpoints)
+      retriever.wait() # Takes a bit of time to get and parse the ldap information
+      targets = retriever.GetExecutionTargets()
+      ceStats = targets[0].ComputingShare
+      gLogger.debug("Running jobs for CE %s : %s" % (self.ceHost, ceStats.RunningJobs))
+      gLogger.debug("Waiting jobs for CE %s : %s" % (self.ceHost, ceStats.WaitingJobs))
+      result['RunningJobs'] = ceStats.RunningJobs
```

Comment:
Not sure ... as far as I can see, it works as it is currently. Also see lines 249, 250 in ComputingElement.py?

https://github.com/DIRACGrid/DIRAC/blob/integration/Resources/Computing/ComputingElement.py#L249

I will wait for your response on this. The rest of your comments are "of course - what was I thinking!" :)